### PR TITLE
[aenetworks] Prevent missing ratings extractor crash

### PR DESCRIPTION
**Pull request by agibson-fl**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Certain shows without ratings on AE networks will crash and not allow the download to complete.
To duplicate this:
`yt-dlp -V https://play.history.com/shows/kennedy/season-1/episode-2`
or 
`yt-dlp -V https://play.history.com/shows/kennedy/season-1/episode-3`
(not sure if one will get ratings in the meantime of this being merged and prevent duplication)

I found code already in this project from the amcnetworks.py extractor that fixes the problem there. I copied that code over to the aenetworks.py and tested it manually. It works.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/aenetworks.py [hunk 1]
```diff
@@ -90,10 +91,11 @@ def _extract_aetn_info(self, domain, filter_key, filter_value, url):
         info = self._parse_theplatform_metadata(theplatform_metadata)
         auth = None
         if theplatform_metadata.get('AETN$isBehindWall'):
+            rating = try_get(theplatform_metadata, lambda x: x['ratings'][0]['rating'])
```
Comment (bashonly) [inline yt_dlp/extractor/aenetworks.py:94]:
```suggestion
        if theplatform_metadata.get('AETN$isBehindWall'):
```

File: yt_dlp/extractor/aenetworks.py [hunk 2]
```diff
@@ -90,10 +91,11 @@ def _extract_aetn_info(self, domain, filter_key, filter_value, url):
...
             resource = self._get_mvpd_resource(
                 requestor_id, theplatform_metadata['title'],
                 theplatform_metadata.get('AETN$PPL_pplProgramId') or theplatform_metadata.get('AETN$PPL_pplProgramId_OLD'),
-                theplatform_metadata['ratings'][0]['rating'])
+                rating)
```
Comment (bashonly) [inline yt_dlp/extractor/aenetworks.py:98]:
```suggestion
                traverse_obj(theplatform_metadata, ('ratings', 0, 'rating')))
```

File: yt_dlp/extractor/aenetworks.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
     int_or_none,
     remove_start,
     traverse_obj,
+    try_get,
```
Comment (bashonly) [inline yt_dlp/extractor/aenetworks.py:8]:
```suggestion
    traverse_obj,
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Comment by agibson-fl** [comment]
@bashonly I pushed the requested changes.

**Pull request merged**